src/metabase/driver/hive_like.clj

```diff
@@ -0,0 +1,149 @@
+    :day (trunc-with-format "yyyy-MM-dd" (hx/->timestamp expr))
+    :day-of-week (hx/->integer (date-format "u"
+                                            (hx/+ (hx/->timestamp expr)
+                                                  (hsql/raw "interval '1' day"))))
+    :day-of-month (hsql/call :dayofmonth (hx/->timestamp expr))
+    :day-of-year (hx/->integer (date-format "D" (hx/->timestamp expr)))
+    :week (hsql/call :date_sub
+                     (hx/+ (hx/->timestamp expr)
+                           (hsql/raw "interval '1' day"))
+                     (date-format "u"
+                                  (hx/+ (hx/->timestamp expr)
+                                        (hsql/raw "interval '1' day"))))
+    :week-of-year (hsql/call :weekofyear (hx/->timestamp expr))
+    :month (hsql/call :trunc (hx/->timestamp expr) (hx/literal :MM))
+    :month-of-year (hsql/call :month (hx/->timestamp expr))
+    :quarter (hsql/call :add_months
+                        (hsql/call :trunc (hx/->timestamp expr) (hx/literal :year))
+                        (hx/* (hx/- (hsql/call :quarter (hx/->timestamp expr))
+                                    1)
+                              3))
+    :quarter-of-year (hsql/call :quarter (hx/->timestamp expr))
+    :year (hsql/call :year (hx/->timestamp expr))))
+
+(defn date-interval
+  "Returns a SQL expression to calculate a time interval using the Hive SQL dialect"
+  [unit amount]
+  (hsql/raw (format "(NOW() + INTERVAL '%d' %s)" (int amount) (name unit))))
+
+(defn string-length-fn
+  "A SQL function call that returns the string length of `field-key`"
+  [field-key]
+  (hsql/call :length field-key))
+
+;; ignore the schema when producing the identifier
+(defn qualified-name-components
+  "Return the pieces that represent a path to FIELD, of the form `[table-name parent-fields-name* field-name]`.
+   This function should be used by databases where schemas do not make much sense."
+  [{field-name :name, table-id :table_id, parent-id :parent_id}]
+  (conj (vec (if-let [parent (metabase.models.field/Field parent-id)]
+               (qualified-name-components parent)
+               (let [{table-name :name, schema :schema} (db/select-one ['Table :name :schema], :id table-id)]
+                 [table-name])))
+        field-name))
+
+(defn field->identifier
+  "Returns an identifier for the given field"
+  [field]
+  (apply hsql/qualify (qualified-name-components field)))
+
+(defn- run-query
+  "Run the query itself."
+  [{sql :query, params :params, remark :remark} connection]
+  (let [sql              (str "-- " remark "\n" (hx/unescape-dots sql))
+        statement        (into [sql] params)
+        [columns & rows] (jdbc/query connection statement {:identifiers identity, :as-arrays? true})]
+    {:rows    (or rows [])
+     :columns columns}))
+
+(defn run-query-without-timezone
+  "Runs the given query without trying to set a timezone"
+  [driver settings connection query]
+  (run-query query connection))
```

Comment:
It looks like these are copied over from the implementations in `generic-sql.query-processor` but you just took the timezone stuff out. Those functions are supposed to work for databases that don't support setting a timezone (defined by whether they declare the `:set-timezone` feature). What was the issue you were having with them? If you couldn't get them to work it's probably better we tweak them so we can reuse them instead of copying them to change one tiny thing

Reply:
The big issue is the use of transactions, not the timezone thing. I tried modifying the code to take `transaction?` as an argument, but I didn't like the result. I think it should probably be in the driver features.